targets/arm/mikroe/sam/src/tim/implementations/implementation_1/hal_ll_tim.c

```diff
@@ -0,0 +1,752 @@
+/*!
+ * @file  hal_ll_tim.c
+ * @brief TIM HAL LOW LEVEL layer implementation.
+ */
+
+#include "hal_ll_gpio.h"
+#include "hal_ll_pmc.h"
+#include "hal_ll_tim.h"
+#include "hal_ll_tim_pin_map.h"
+
+/*!< @brief Local handle list */
+static volatile hal_ll_tim_handle_register_t hal_ll_module_state[ TIM_MODULE_COUNT ];
+
+// ------------------------------------------------------------- PRIVATE MACROS
+/*!< @brief Register bits macros. */
+#define HAL_LL_TC_CCR_CLKEN_BIT     0
+#define HAL_LL_TC_CCR_CLKDIS_BIT    1
+#define HAL_LL_TC_CCR_SWTRG_BIT     2
+#define HAL_LL_TC_EMR_NODIVCLK_BIT  8
+#define HAL_LL_TC_CMR_WAVE_BIT      15
+#define HAL_LL_PID_TC6_BIT          15
+#define HAL_LL_PID_TC7_BIT          16
+#define HAL_LL_PID_TC8_BIT          17
+#define HAL_LL_PID_TC9_BIT          18
+#define HAL_LL_PID_TC10_BIT         19
+#define HAL_LL_PID_TC11_BIT         20
+#define HAL_LL_PID_TC0_BIT          23
+#define HAL_LL_PID_TC1_BIT          24
+#define HAL_LL_PID_TC2_BIT          25
+#define HAL_LL_PID_TC3_BIT          26
+#define HAL_LL_PID_TC4_BIT          27
+#define HAL_LL_PID_TC5_BIT          28
+
+/*!< @brief Register masks macros. */
+#define HAL_LL_TC_CMR_TIMER_CLOCK2_MASK 0x1
+#define HAL_LL_TC_CMR_TIMER_CLOCK3_MASK 0x2
+#define HAL_LL_TC_CMR_TIMER_CLOCK4_MASK 0x3
+#define HAL_LL_TC_CMR_EEVT_XC0_MASK     0x0400
+#define HAL_LL_TC_CMR_WAVSEL_UP_RC_MASK 0x4000
+#define HAL_LL_TC_CMR_ACPA_CLEAR_MASK   0x20000
+#define HAL_LL_TC_CMR_ACPC_SET_MASK     0x40000
+#define HAL_LL_TC_CMR_BCPB_CLEAR_MASK   0x2000000
+#define HAL_LL_TC_CMR_BCPC_SET_MASK     0x4000000
+
+/*!< @brief Helper macro fpr getting clock speed */
+#define hal_ll_get_system_clock (Get_Fosc_kHz() * 1000)
+#define hal_ll_tim_timer_clock_2 (hal_ll_get_system_clock / 8)
+#define hal_ll_tim_timer_clock_3 (hal_ll_get_system_clock / 32)
+#define hal_ll_tim_timer_clock_4 (hal_ll_get_system_clock / 128)
+
+/*!< @brief Helper macro for getting module specific control register structure */
+#define hal_ll_tim_get_base_struct(_handle) ((hal_ll_tim_base_handle_t *)_handle)
+
+/*!< @brief Helper macro for getting hal_ll_module_state address */
+#define hal_ll_tim_get_module_state_address ((hal_ll_tim_handle_register_t *)*handle)
+
+/*!< @brief Helper macro for getting module specific control register structure base address // first register address */
+#define hal_ll_tim_get_handle ( hal_ll_tim_handle_register_t * )hal_ll_tim_get_module_state_address->hal_ll_tim_handle
+
+/*!< @brief Helper macro for getting module specific base address directly from HAL layer handle */
+#define hal_ll_tim_get_base_from_hal_handle ((hal_ll_tim_hw_specifics_map_t *)((hal_ll_tim_handle_register_t *)\
+                                            (((hal_ll_tim_handle_register_t *)(handle))->hal_ll_tim_handle))->hal_ll_tim_handle)->base
+
+// -------------------------------------------------------------- PRIVATE TYPES
+/*!< @brief TIM channel register ctructure. */
+typedef struct
+{
+    uint32_t ccr;
```

Comment:
Check file for appropriate use of `hal_ll_base_addr_t` and `hal_ll_pin_name_t`.

Reply:
Changed it for all source files where needed.